cmd/analyzetestresults/analyzetestresults.go

```diff
@@ -0,0 +1,221 @@
+package analyzetestresults
+
+import (
+	"encoding/xml"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/bsm/ginkgo/v2/reporters"
+	"github.com/konflux-ci/qe-tools/pkg/oci"
+	"k8s.io/klog/v2"
+
+	"github.com/konflux-ci/qe-tools/pkg/types"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// FailureType collects the types of failures met in the PipelineRun
+type FailureType string
+
+const (
+	dropdownSummaryString = "Click to view logs"
+
+	// OtherFailure represents the type failure that hasn't been identified by the analyzer
+	OtherFailure FailureType = "otherFailure"
+	// ClusterCreationFailure represents the issue with provisioning a test cluster
+	ClusterCreationFailure FailureType = "clusterCreationFailure"
+	// TestRunFailure represents the issue with provisioning a test cluster
+	TestRunFailure FailureType = "testRunFailure"
+	// TestCaseFailure represents the issue with provisioning a test cluster
+	TestCaseFailure FailureType = "testCaseFailure"
+)
+
+var (
+	ociArtifactRef              string
+	clusterProvisionLogFilename string
+	jUnitFilename               string
+	e2eTestRunLogFilename       string
+	outputFilename              string
+)
+
+// FailedTestCasesReport collects the data about failures
+type FailedTestCasesReport struct {
+	headerString        string
+	failedTestCaseNames []string
+
+	clusterProvisionLog string
+	e2eTestLog          string
+
+	failureType FailureType
+}
```

Comment:
I would like to see this in a new like pkg/testresults/results.go or something like that. I can see this potentially use as struct to capture flaky test information and to report on flaky tests in the future.